src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -552,7 +553,12 @@ internal sealed override CommandLineArguments CommonParse(IEnumerable<string> ar
                             continue;
 
                         case "nosdkpath":
-                            sdkDirectory = null;
+                            noSdkPath = true;
+
+                            continue;
+
+                        case "sdkpath":
```

Comment:
Overall I'm a bit nervous about introducing a new command line argument here. This is a problem that we previously solved by changing how the compiler was created. Did you consider making sdk path another value in `BuildRequest`? That already carries along information like working directory, temp directory, etc ... Seems like sdk directory could easily be another value there. That would remove the need for a new command line parameter, remove the need to solve conflicting arguments and, seemingly, also get vbc and csc in sync. Is there something I'm missing with that?